.github/workflows/master.yml

```diff
@@ -462,6 +463,42 @@ jobs:
         - 14.x
         pythonversion:
         - "3.7"
+  generate_coverage_data:
+    name: generate_coverage_data
+    needs: prerequisites
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+    - name: Generate Coverage Data
+      run: |
+        echo "Coverage output directory: $COVERAGE_OUTPUT_DIR"
+        make tfgen
+  upload_coverage_data:
+    name: upload_coverage_data
+    needs: generate_coverage_data
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+    - name: Obtain current date & S3 address
+      run: |
+        currDateTime=$(date +"%Y-%m-%d_%H-%M-%S")
```

Comment:
if any of these are environment variables then we need to make sure they are treated as such - the results of a script is not necessarily going to carry variables to the next step. So anything that needs to be reused elsewhere needs to be made an env var via use of GITHUB_ENV